process/interceptors/processor/sovereignHdrInterceptorProcessor.go

```diff
@@ -0,0 +1,72 @@
+package processor
+
+import (
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/hashing"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+// ArgsSovereignHeaderInterceptorProcessor is a struct placeholder used to create a new sovereign extended header interceptor processor
+type ArgsSovereignHeaderInterceptorProcessor struct {
+	BlockBlackList           process.TimeCacher
+	Hasher                   hashing.Hasher
+	Marshaller               marshal.Marshalizer
+	IncomingHeaderSubscriber process.IncomingHeaderSubscriber
+}
+
+type sovereignHeaderInterceptorProcessor struct {
+	blackList                process.TimeCacher
+	Hasher                   hashing.Hasher
+	Marshaller               marshal.Marshalizer
+	IncomingHeaderSubscriber process.IncomingHeaderSubscriber
+}
+
+// NewSovereignHdrInterceptorProcessor creates a new sovereign extended header interceptor processor
+func NewSovereignHdrInterceptorProcessor(argument *ArgsSovereignHeaderInterceptorProcessor) (*sovereignHeaderInterceptorProcessor, error) {
+	if argument == nil {
+		return nil, process.ErrNilArgumentStruct
+	}
+	if check.IfNil(argument.BlockBlackList) {
+		return nil, process.ErrNilBlackListCacher
+	}
+
+	return &sovereignHeaderInterceptorProcessor{
+		blackList:                argument.BlockBlackList,
+		Hasher:                   argument.Hasher,
+		Marshaller:               argument.Marshaller,
+		IncomingHeaderSubscriber: argument.IncomingHeaderSubscriber,
+	}, nil
+}
+
+// Validate checks if the intercepted data can be processed
+func (hip *sovereignHeaderInterceptorProcessor) Validate(data process.InterceptedData, _ core.PeerID) error {
+	_, ok := data.(process.ExtendedHeaderValidatorHandler)
+	if !ok {
+		return fmt.Errorf("sovereignHeaderInterceptorProcessor.Validate error: %w", process.ErrWrongTypeAssertion)
+	}
+
+	return nil
+}
+
+// Save will save the received data into headers pool
+func (hip *sovereignHeaderInterceptorProcessor) Save(data process.InterceptedData, _ core.PeerID, _ string) error {
+	_, ok := data.(process.ExtendedHeaderValidatorHandler)
+	if !ok {
+		return fmt.Errorf("sovereignHeaderInterceptorProcessor.Save error: %w", process.ErrWrongTypeAssertion)
+	}
+
+	return nil
```

Comment:
Here should be added more code for saving, right ? If so, you can put a TODO here!